src/distributions/mod.rs

```diff
@@ -275,42 +276,30 @@ impl<'a, D, R, T> Iterator for DistIter<'a, D, R, T>
 /// let mut rng = thread_rng();
 /// let erased_rng: &mut RngCore = &mut rng;
 /// let val: f32 = erased_rng.sample(Standard);
-/// println!("f32 from (0, 1): {}", val);
+/// println!("f32 from [0, 1): {}", val);
 /// ```
 ///
-/// # Open interval for floats
-/// In theory it is possible to choose between an open interval `(0, 1)`, and
-/// the half-open intervals `[0, 1)` and `(0, 1]`. All can give a distribution
-/// with perfectly uniform intervals. Many libraries in other programming
-/// languages default to the closed-open interval `[0, 1)`. We choose here to go
-/// with *open*, with the arguments:
+/// # Floating point implementation
+/// The floating point implementations for `Standard` generate a random value in
+/// the half-open interval [0, 1).
 ///
-/// - The chance to generate a specific value, like exactly 0.0, is *tiny*. No
-///   (or almost no) sensible code relies on an exact floating-point value to be
-///   generated with a very small chance (1 in 2<sup>23</sup> (approx. 8
-///   million) for `f32`, and 1 in 2<sup>52</sup> for `f64`). What is relied on
-///   is having a uniform distribution and a mean of `0.5`.
-/// - Several common algorithms rely on never seeing the value `0.0` generated,
-///   i.e. they rely on an open interval. For example when the logarithm of the
-///   value is taken, or used as a devisor.
+/// All values that can be generated are multiples of ε/2. For `f32` the 23 most
+/// significant random bits of an `u32` are used, for `f64` 53 from an `u64`.
+/// The conversion uses the common multiply-based approach.
 ///
-/// In other words, the guarantee some value *could* be generated is less useful
-/// than the guarantee some value (`0.0`) is never generated. That makes an open
-/// interval a nicer choice.
+/// The `Open01` distribution provides an alternative: it generates values in
+/// the open interval (0, 1), with one less random bit. It uses the slightly
+/// faster transmute-based approach for the conversion to a floating point
+/// value.
```

Comment:
"Slightly faster" isn't really correct; I just benchmarked and found no difference. You could however say *(which may be faster on some architectures)*.